### PR TITLE
Add platform test fixes

### LINKED ISSUE #9582: Exception when weights array for loss function is a scalar
I stumbled upon this exception:

```
Caused by: java.lang.IllegalArgumentException: Weights array must be a row vector
	at org.nd4j.linalg.lossfunctions.impl.LossBinaryXENT.<init>(LossBinaryXENT.java:100)
	at org.nd4j.linalg.lossfunctions.impl.LossBinaryXENT.<init>(LossBinaryXENT.java:72)
...
```

with weights array being a scalar. Dl4j does the following test:

```
   public LossBinaryXENT(@JsonProperty("clipEps") double clipEps, @JsonProperty("weights") INDArray weights){
        if (weights != null && !weights.isRowVector()) {
            throw new IllegalArgumentException("Weights array must be a row vector");
        }
...
```

Shouldn't it be sufficient to test on "isRowVectorOrScalar"?

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Adds kotlin compilation and fixes misc test issues with model import.
Fixes https://github.com/eclipse/deeplearning4j/issues/9582

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [X ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [X ] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/executioner/DefaultOpExecutioner.java [hunk 1]
```diff
@@ -268,6 +268,8 @@ public void setProfilingMode(ProfilingMode mode) {
                 config = ProfilerConfig.builder().build();
                 break;
         }
+
+        System.nanoTime();
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/executioner/DefaultOpExecutioner.java]: This looks like it is here by mistake.

File: platform-tests/pom.xml [hunk 2]
```diff
@@ -1,277 +1,348 @@
...
+            </snapshots>
+        </repository>
+    </repositories>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.7</maven.compiler.source>
+        <maven.compiler.target>1.7</maven.compiler.target>
+        <dl4j.version>1.0.0-SNAPSHOT</dl4j.version>
+        <platform.classifier>windows-x86_64</platform.classifier>
```
Comment (treo) [inline platform-tests/pom.xml:49]: Did you forget to change this back, or is this intentionally tied to windows?

File: platform-tests/pom.xml [hunk 3]
```diff
@@ -1,277 +1,348 @@
...
+                <groupId>org.jetbrains.kotlin</groupId>
+                <artifactId>kotlin-maven-plugin</artifactId>
+                <version>${kotlin.version}</version>
+                <configuration>
+                    <args>
+                        <arg>-Xjsr305=strict</arg>
+                    </args>
+                    <compilerPlugins>
+                        <plugin>spring</plugin>
+                        <plugin>jpa</plugin>
```
Comment (treo) [inline platform-tests/pom.xml:265]: Why the spring and jpa plugins?

File: platform-tests/src/test/kotlin/org/nd4j/samediff/frameworkimport/onnx/loader/TestOnnxProcessLoader.kt [hunk 4]
```diff
@@ -37,12 +37,15 @@ class TestOnnxProcessLoader {
...
+        System.setProperty("org.bytedeco.javacpp.nopointergc","true")
+        Nd4j.getExecutioner().enableDebugMode(true)
+        Nd4j.getExecutioner().enableVerboseMode(true)
         val importer = OnnxFrameworkImporter()
         val resource = ClassPathResource("mobilenet.onnx").file
         val inputs = Nd4j.ones(1,3,224,224)
         val inputDict = mapOf("input.1" to inputs)
         val runImport = importer.runImport(resource.absolutePath,inputDict)
-
+        println("Ran import")
```
Comment (treo) [inline platform-tests/src/test/kotlin/org/nd4j/samediff/frameworkimport/onnx/loader/TestOnnxProcessLoader.kt]: I guess this was left over from a debugging session?

**Review by treo — CHANGES_REQUESTED** [review]

File: platform-tests/pom.xml (hunk 2 above)
Comment (agibsonccc) [reply platform-tests/pom.xml:49]: @treo good point. Generally it's expected that this is specified on command line. The default is really anything that we need it to be. I cleaned up the dup maven.compiler/source target and changed this back to linux.

File: platform-tests/pom.xml (hunk 3 above)
Comment (agibsonccc) [reply platform-tests/pom.xml:265]: Left over from boilerplate.

File: platform-tests/src/test/kotlin/org/nd4j/samediff/frameworkimport/onnx/loader/TestOnnxProcessLoader.kt (hunk 4 above)
Comment (agibsonccc) [reply platform-tests/src/test/kotlin/org/nd4j/samediff/frameworkimport/onnx/loader/TestOnnxProcessLoader.kt]: Yeah.

File: platform-tests/src/test/java/org/nd4j/imports/tfgraphs/TFGraphTestAllSameDiff.java [hunk 5]
```diff
@@ -199,6 +205,7 @@ public void testOutputOnly(Map<String, INDArray> inputs, Map<String, INDArray> p
             Nd4j.getExecutioner().enableDebugMode(true);
             Nd4j.getExecutioner().enableVerboseMode(true);
             TFGraphTestAllHelper.checkOnlyOutput(inputs, predictions, modelName, BASE_DIR, MODEL_FILENAME, EXECUTE_WITH, TFGraphTestAllHelper.LOADER, maxRE, minAbs, verboseDebugMode);
+            System.out.println();
```
Comment (treo) [inline platform-tests/src/test/java/org/nd4j/imports/tfgraphs/TFGraphTestAllSameDiff.java]: Why the empty println here?

**Review by treo — CHANGES_REQUESTED** [review]

**Review by treo — APPROVED** [review]

**Pull request merged**